narayana-spring-boot-core/src/main/java/me/snowdrop/boot/narayana/core/jms/GenericXAConnectionFactoryWrapper.java

```diff
@@ -32,18 +33,31 @@
     /**
      * Create a new {@link GenericXAConnectionFactoryWrapper} instance.
      *
-     * @param transactionManager underlying transaction manager
-     * @param xaRecoveryModule   recovery module to register data source with.
-     * @param properties         Narayana properties
+     * @param transactionManager  underlying transaction manager
+     * @param xaRecoveryModule    recovery module to register data source with.
+     * @param properties          MessagingHub properties
      */
     public GenericXAConnectionFactoryWrapper(TransactionManager transactionManager, XARecoveryModule xaRecoveryModule,
-            NarayanaProperties properties) {
-        super(xaRecoveryModule, properties);
+            MessagingHubConnectionFactoryProperties properties) {
+        this(transactionManager, xaRecoveryModule, properties, RecoveryCredentialsProperties.DEFAULT);
+    }
+
+    /**
+     * Create a new {@link GenericXAConnectionFactoryWrapper} instance.
+     *
+     * @param transactionManager  underlying transaction manager
+     * @param xaRecoveryModule    recovery module to register data source with.
+     * @param properties          MessagingHub properties
+     * @param recoveryCredentials Credentials for recovery helper
+     */
+    public GenericXAConnectionFactoryWrapper(TransactionManager transactionManager, XARecoveryModule xaRecoveryModule,
+            MessagingHubConnectionFactoryProperties properties, RecoveryCredentialsProperties recoveryCredentials) {
```

Comment:
Pooled wrapper properties shouuld not be here

Reply:
Right, I'll remove it.